alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -90,9 +90,10 @@ public ProviderProject saveProject(final String providerName, final ProviderProj
     }
 
     @Override
-    public List<ProviderProject> deleteAndSaveAllProjects(final String providerName, final Collection<ProviderProject> providerProjects) {
-        providerProjectRepository.deleteAllByProvider(providerName);
-        final Iterable<ProviderProjectEntity> providerProjectEntities = providerProjects
+    public List<ProviderProject> deleteAndSaveAllProjects(final String providerName, final Collection<ProviderProject> providerProjectsToRemove, final Collection<ProviderProject> providerProjectsToAdd) {
```

Comment:
Should this be split into 2 separate methods? We can still have this method, but having the 2 functionalities split out as well makes more sense to me.

Reply:
Then would you also want the same done for Users, which does the same?

Reply:
If Users also has a method that does multiple things like this, then yes

Reply:
We should also remove "all" from the method name since it is no longer touching every single project.